WordPress/gutenberg
# Merge Comments and Post Comments blocks

Tracking issue: https://github.com/WordPress/gutenberg/issues/41451

## What?

Merge the Comments block into the Post Comments block. This PR is WIP and continues the work in https://github.com/WordPress/gutenberg/pull/40522.

The idea is to end up with a single block comprising both blocks:
- Comments as the default behavior
- Post Comments as a fallback

More info in https://github.com/WordPress/gutenberg/issues/40521.

We could also take this opportunity to change the Comments block ID from `core/comments-query-loop` to `core/comments` and end up with something like this:

![image](https://user-images.githubusercontent.com/6917969/174559409-9fb914d2-3070-407e-bcdb-f80128d691e9.png)

## Why?

See https://github.com/WordPress/gutenberg/pull/40522#issuecomment-1109591199

## How?

The block differentiates between the "legacy" PHP mode and the editable mode using an attribute called `legacy`. That property is `false` by default, and it's only set to `true` when converting the old Post Comments into the Comments block. The block conversion is automatically done when you open the WP editor.

The block is either static or dynamic depending on the value of `legacy` (`false` or `true`, respectively).

## Testing Instructions
 
1. Check out this branch.
2. With a block theme installed that uses the Post Comment block in its Single template, visit a post with comments.
3. Ensure that the Post Comments block renders as usual.
4. Move to the Site Editor and open the Single template.
5. Check that the block has turned into a Comments block. It should show a placeholder and a warning message to switch the block to the editable version.
6. Open the code editor and verify that the block is now ``. Close the code editor.
7. Save changes (you would need to make minor changes, like reordering blocks).
8. Revisit the previous post. It should look exactly as before (i.e., the PHP version).
9. Go to the Site Editor again, and now click the `switch` button. Now, inner blocks should appear.
10. Open the code editor. The block should not have the `legacy` attribute (it is `false` by default) and should contain rendered markup and inner blocks.
11. Save and visit the post again. The new version should render.

Other things to test:
- Post Comments should not appear in the block inserter.
- Adding the Comments block should show the editable version by default.
